src/NServiceBus.Transport.Msmq/DelayedDelivery/DueDelayedMessagePoller.cs

```diff
@@ -0,0 +1,341 @@
+namespace NServiceBus.Transport.Msmq.DelayedDelivery
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Logging;
+    using Routing;
+    using Unicast.Queuing;
+
+    class DueDelayedMessagePoller
+    {
+        public DueDelayedMessagePoller(
+            MsmqMessageDispatcher dispatcher,
+            IDelayedMessageStore delayedMessageStore,
+            int numberOfRetries,
+            Action<string, Exception, CancellationToken> criticalErrorAction,
+            string timeoutsErrorQueue,
+            Dictionary<string, string> faultMetadata,
+            TransportTransactionMode transportTransactionMode,
+            TimeSpan timeToTriggerFetchCircuitBreaker,
+            TimeSpan timeToTriggerDispatchCircuitBreaker,
+            int maximumRecoveryFailuresPerSecond
+        )
+        {
+            txOption = transportTransactionMode == TransportTransactionMode.TransactionScope
+                ? TransactionScopeOption.Required
+                : TransactionScopeOption.RequiresNew;
+            this.delayedMessageStore = delayedMessageStore;
+            errorQueue = timeoutsErrorQueue;
+            this.faultMetadata = faultMetadata;
+            this.numberOfRetries = numberOfRetries;
+            this.dispatcher = dispatcher;
+            fetchCircuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MsmqDelayedMessageFetch", timeToTriggerFetchCircuitBreaker,
+                ex => criticalErrorAction("Failed to fetch due delayed messages from the storage", ex, CancellationToken.None));
+
+            dispatchCircuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MsmqDelayedMessageDispatch", timeToTriggerDispatchCircuitBreaker,
+                ex => criticalErrorAction("Failed to dispatch delayed messages to destination", ex, CancellationToken.None));
+
+            failureHandlingCircuitBreaker = new FailureRateCircuitBreaker("MsmqDelayedMessageFailureHandling", maximumRecoveryFailuresPerSecond, ex => criticalErrorAction("Failed to execute error handling for delayed message forwarding", ex, CancellationToken.None));
```

Comment:
To copy the same stile as the other circuit breakers.
```suggestion
            failureHandlingCircuitBreaker = new FailureRateCircuitBreaker("MsmqDelayedMessageFailureHandling", maximumRecoveryFailuresPerSecond,
                ex => criticalErrorAction("Failed to execute error handling for delayed message forwarding", ex, CancellationToken.None));
```

Reply:
👍